source/slang/type-layout.cpp

```diff
@@ -109,15 +119,15 @@ struct DefaultLayoutRulesImpl : SimpleLayoutRulesImpl
         return structInfo;
     }
 
-    size_t AddStructField(UniformLayoutInfo* ioStructInfo, UniformLayoutInfo fieldInfo) override
+    LayoutSize AddStructField(UniformLayoutInfo* ioStructInfo, UniformLayoutInfo fieldInfo) override
```

Comment:
Can this method return infinite? Lots of use sites, appear to require that it's finite.

Reply:
This change mostly just handles the case of unbounded arrays of resource types, and ignores the issue of unbounded arrays of uniform data. That is allowed in GLSL in certain contexts, with rules similar to what you have in C compilers that allow an unsized array as the last field in a `struct` type.

Long story short, this method can return infinity, but if it does that indicates an error in the user's code (they tried to put a field *after* something that requires unbounded storage). We should be properly detecting and diagnosing such cases, but that is out of scope for the current change so the default assert-failure behavior in `getFiniteValue()` will have to do.